codec-http2/src/test/java/io/netty/handler/codec/http2/Http2MultiplexCodecTransportTest.java

```diff
@@ -0,0 +1,138 @@
+        ServerBootstrap sb = new ServerBootstrap();
+        sb.group(eventLoopGroup);
+        sb.channel(NioServerSocketChannel.class);
+        sb.childHandler(new ChannelInitializer<Channel>() {
+            @Override
+            protected void initChannel(Channel ch) {
+                ch.pipeline().addLast(Http2MultiplexCodecBuilder.forServer(new HttpInboundHandler()).build());
+                ch.pipeline().addLast(new ChannelInboundHandlerAdapter() {
+                    @Override
+                    public void channelActive(ChannelHandlerContext ctx) {
+                        serverConnectedChannelRef.set(ctx.channel());
+                        serverConnectedChannelLatch.countDown();
+                    }
+
+                    @Override
+                    public void channelRead(ChannelHandlerContext ctx, Object msg) {
+                        if (msg instanceof Http2SettingsAckFrame) {
+                            serverAckLatch.countDown();
+                        }
+                    }
+                });
+            }
+        });
+        serverChannel = sb.bind(new InetSocketAddress(NetUtil.LOCALHOST, 0)).awaitUninterruptibly().channel();
+
+        Bootstrap bs = new Bootstrap();
+        bs.group(eventLoopGroup);
+        bs.channel(NioSocketChannel.class);
+        bs.handler(new ChannelInitializer<Channel>() {
+            @Override
+            protected void initChannel(Channel ch) {
+                ch.pipeline().addLast(Http2MultiplexCodecBuilder
+                        .forClient(new HttpInboundHandler()).autoAckSettings(false).build());
+                ch.pipeline().addLast(new ChannelInboundHandlerAdapter() {
+                    @Override
+                    public void channelRead(ChannelHandlerContext ctx, Object msg) {
+                        if (msg instanceof Http2SettingsFrame) {
+                            clientSettingsLatch.countDown();
+                        }
+                    }
+                });
+            }
+        });
+        clientChannel = bs.connect(serverChannel.localAddress()).awaitUninterruptibly().channel();
+        serverConnectedChannelLatch.await();
+        serverConnectedChannel = serverConnectedChannelRef.get();
+
+        serverConnectedChannel.writeAndFlush(new DefaultHttp2SettingsFrame(new Http2Settings()
+                .maxConcurrentStreams(10)));
+
+        clientSettingsLatch.await();
+
+        // We expect a timeout here because we want to asynchronously generate the SETTINGS ACK below.
+        assertFalse(serverAckLatch.await(300, MILLISECONDS));
+
+        // We expect 2 settings frames, the initial settings frame during connection establishment and the setting frame
+        // written in this test. We should ack both of these settings frames.
+        clientChannel.write(DefaultHttp2SettingsAckFrame.INSTANCE);
+        clientChannel.writeAndFlush(DefaultHttp2SettingsAckFrame.INSTANCE);
```

Comment:
should we sync on the future ?

Reply:
We don't have to because we wait for the message to be received on the server side below with `serverAckLatch`.

Reply:
sure but if we `sync` we will get the right exception in case of a error while writing.

Reply:
Same comment for others writes btw

Reply:
ok I can add the sync